BookBuilder.py

```diff
@@ -349,54 +359,32 @@ def iterator(self, chapter, openingName):
             if lineCount == 0:
                 printerFinalLine.append(line) #we add line to go to print
             else:
-                logging.debug("duplicate line ", line)
+                logging.debug(f"duplicate line {line}")
             logging.debug(f"final line count { lineCount+1 } for line {lineString}")
-        
-        
-        logging.debug (f'we sort the lines by consecutive move probabilities')        
+
+        logging.debug(f'we sort the lines by consecutive move probabilities')
+
         def extract_key(printerFinalLine):
             return [v for _, v in printerFinalLine[2]]
 
         printerFinalLine = sorted(printerFinalLine, key=extract_key)
         
         for line in printerFinalLine:
-            logging.debug (line[0])        
-
-
-        
+            logging.debug (line[0])
         
-        logging.debug (f'we reverse the sort to make long to short')        
-        if config.LONGTOSHORT == 1:
+        logging.debug (f'we reverse the sort to make long to short')
+        if self.settings.book.order.LONG_TO_SHORT:
             printerFinalLine.reverse() #we make the longest (main lines) first
-            
         
         for line in printerFinalLine:
             logging.debug (line[0])
-            
-            
-        
         
         #we print the final list of lines
         logging.debug(f'number of final lines {len(printerFinalLine)}')
         logging.debug(f'final line sorted {printerFinalLine}')
-        printer = Printer( f"{working_dir}/Chapter_{chapter}_{openingName}.pgn")
+        printer = Printer(self.settings, f"{working_dir}/Chapter_{chapter}_{openingName}.pgn")
 
         lineNumber = 1        
         for pgn, cumulative, likelyPath, winRate, Games in printerFinalLine:
             printer.print(pgn, cumulative, likelyPath, winRate, Games, lineNumber, openingName)
             lineNumber += 1
-
-
-def main():

```

Comment:
Moved to the main.py file
Instead of running the main logic immediately, grower.run() is now only called after a user clicks the "Generate PGN" button